src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -76,7 +77,10 @@ public ReadOnlyTaskList getTaskList() {
     @Override
     public ObservableList<Task> getFilteredTaskList() {
         ObservableList<Task> tasklist = model.getFilteredTaskList();
-        tasklist.forEach((task) -> task.triggerRecurringIfPresent(model));
+        for (int i = 0; i < tasklist.size(); i++) {
+            Task currentTask = tasklist.get(i);
```

Comment:
Also if i call list multiple times, repeated timers will be setup and this could lead to some bugs. A task could be marked as undone many times